apis/v1alpha1/clientsettingspolicy_types.go

```diff
@@ -0,0 +1,149 @@
+type ClientBody struct {
+	// MaxSize sets the maximum allowed size of the client request body.
+	// If the size in a request exceeds the configured value,
+	// the 413 (Request Entity Too Large) error is returned to the client.
+	// Setting size to 0 disables checking of client request body size.
+	// Default: 1m.
+	//
+	// +optional
+	MaxSize *Size `json:"maxSize,omitempty"`
+
+	// Timeout defines a timeout for reading client request body. The timeout is set only for a period between
+	// two successive read operations, not for the transmission of the whole request body.
+	// If a client does not transmit anything within this time, the request is terminated with the
+	// 408 (Request Time-out) error.
+	// Default: 60s.
+	//
+	// +optional
+	Timeout *Duration `json:"timeout,omitempty"`
+}
+
+// ClientKeepAlive defines the keep-alive settings for clients.
+type ClientKeepAlive struct {
+	// Requests sets the maximum number of requests that can be served through one keep-alive connection.
+	// After the maximum number of requests are made, the connection is closed. Closing connections periodically
+	// is necessary to free per-connection memory allocations. Therefore, using too high maximum number of requests
+	// is not recommended as it can lead to excessive memory usage.
+	// Default: 1000.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	Requests *int32 `json:"requests,omitempty"`
+
+	// Time defines the maximum time during which requests can be processed through one keep-alive connection.
+	// After this time is reached, the connection is closed following the subsequent request processing.
+	// Default: 1h.
+	//
+	// +optional
+	Time *Duration `json:"time,omitempty"`
+
+	// Timeout defines the keep-alive timeouts for clients.
+	//
+	// +optional
+	Timeout *ClientKeepAliveTimeout `json:"timeout,omitempty"`
+}
+
+// ClientKeepAliveTimeout defines the timeouts related to keep-alive client connections.
+type ClientKeepAliveTimeout struct {
+	// Server sets the timeout during which a keep-alive client connection will stay open on the server side.
+	// Setting this value to 0 disables keep-alive client connections.
+	// Default: 75s.
+	//
+	// +optional
+	Server *Duration `json:"server,omitempty"`
+
+	// Header sets the timeout in the "Keep-Alive: timeout=time" response header field.
+	//
+	// +optional
+	Header *Duration `json:"header,omitempty"`
+}
+
+// Duration is a string value representing a duration in time.
+// Duration can be specified in milliseconds (ms), seconds (s), minutes (m), hours (h),
+// or days (d). A value without a suffix is seconds.
+// Examples: 300d, 20h, 12m, 45s, 150ms.
+//
+// +kubebuilder:validation:Pattern=`^\d+(ms|s|m|h|d)?$`
+type Duration string
+
+// Size is a string value representing a size. Size can be specified in bytes, kilobytes (k), megabytes (m),
+// or gigabytes (g).
+// Examples: 1024, 8k, 1m.
+//
+// +kubebuilder:validation:Pattern=`^\d+(k|m|g)?$`
+type Size string
```

Comment:
similar to Duration, what are the max sizes here?

Reply:
1000B = 1K
1000K = 1M
1000M = 1G

So also four digits: `^\d{1,4}(k|m|g)?$`?